TRACING.md

```diff
@@ -0,0 +1,272 @@
+  * [Guest journal redirector](#guest-journal-redirector)
+  * [Enable full agent debug](#enable-full-agent-debug)
+
+# Introduction
+
+This document explains agent tracing, which uses 
+the [Jaeger](https://jaegertracing.io) implementation of the [OpenTracing](http://opentracing.io) API.
+
+# OpenTracing summary
+
+An OpenTracing-enabled application creates trace "spans". A span must contain
+the following attributes:
+
+- A name
+- A pair of timestamps (recording the start time and end time of some operation)
+- A reference to the span's parent span
+
+All spans need to be *finished*, or *completed*, to allow the OpenTracing
+framework to generate the final trace information. This implies that the
+agent must be shutdown to obtain a full trace. This will occur after the
+workload has ended.
+
+# Jaeger tracing architecture
+
+The `kata-agent` makes use of the Jaeger
+[golang client package](https://github.com/jaegertracing/jaeger-client-go/).
+
+The Jaeger architecture means that the Jaeger client code running in the
+application reports all span information to a Jaeger agent. This agent
+forwards the information on to other components of a Jaeger system.
+
+The Jaeger client package bound into the application sends the trace spans to
+the Jaeger agent process using the UDP protocol. This is problematic since it
+is not desirable to run a `jaeger-agent` process inside the VM:
+
+- Adding extra binaries and an extra service to run the Jaeger agent inside
+  the VM increases the size of the guest OS image used to
+  create the VM. This is not desirable as the image should be as
+  small as possible.
+
+- Adding a Jaeger agent process into the VM would require
+  special-case code for handling running the `kata-agent`
+  [as the init daemon](https://github.com/kata-containers/osbuilder).
+
+To avoid this complication, agent tracing uses a
+[`VSOCK`](https://www.qemu.org/Features/VirtioVsock) communication channel to
+allow the `kata-agent` to talk to the Jaeger agent running on the host system
+("outside" the VM).
+
+# Guest to Host communication using VSOCK
+
+Since [Jaeger does not yet support VSOCK](https://github.com/jaegertracing/jaeger/issues/1019),
+it is necessary to provide a way for the Jaeger client code (running inside the `kata-agent`
+in the VM), to connect to the Jaeger agent (running on the host). The default
+Jaeger port for such communications is
+[UDP port `6831`](https://www.jaegertracing.io/docs/getting-started/).
+
+The required redirection is a achieved using two instances of the `socat(1)` network utility:
+
+- The first instance runs in the guest environment in the VM and forwards all
+  Jaeger trace traffic out of the VM using a VSOCK socket.
+
+- The second instance runs in the host environment and forwards all Jaeger
+  traffic from the VSOCK socket to the Jaeger agent's UDP port.
+
+Further details:
+
+| Name | Type | Environment | Summary | Details | Explanation |
+|-|-|-|-|-|
```

Comment:
I think you are missing one ` - |` here, which is preventing the table from rendering.